google/ggrc-core
# RP: GGRC-7959/GGRC-7960 Restrict assignee permissions on Assessment pages/tabs

**Should be merged into _GGRC-7772_ branch**



# Issue description

When checkbox "Enable SOX 302 assessment workflow" in the assessment template is switched on, 
all assessments created based on this template shall provide restricted permissions to the assessment assignees. 

1) if assessment is in "Not Started", "In Progress", "Rework needed" or "In Review" statuses:
- assignee cannot edit any built-in, global custom attribute on the assessment
- assignee cannot add/remove labels
- assignee cannot map/unmap any snapshot to assessment
- assignee cannot deprecate assessment
- assignee cannot raise/unmap issue to assessment
- assignee cannot turn on/off linking to issue tracker
+ assignee can add/edit answers to Local Custom Attributes
+ assignee can add/update/delete/reuse evidence url, files
+ assignee can add comments
+ assignee can Undo changes by clicking Undo button in UI
2) if assessment is in "Completed" or "Completed&Verified", "Deprecated" state assignees:
- assignee cannot edit any built-in, global custom attribute on the assessment
- assignee cannot add/remove labels
- assignee cannot map/unmap any snapshot to assessment
- assignee cannot deprecate assessment
- assignee cannot raise/unmap issue to assessment
- assignee cannot turn on/off linking to issue tracker
- assignee cannot move assessment into any other status
- assignee cannot update answers to Local Custom Attributes
- assignees cannot add/update/delete evidence files or evidence urls
- assignees cannot reuse evidence files or evidence urls from related assessments
+ assignee can add comments
+ assignee can Undo changes by clicking Undo button in UI


# Steps to test the changes

Set `is_sox_restricted` to `true` and add properties names to `get_read_only_fields`. Verify that corresponding components are hidden. 

# Solution description

- Registered `if_Sox302_restricted` and `if_edit_is_denied` helpers to hide elements in the stache templates
- Added `isRestricted`, `isSemiRestrictedOnStatus` attributes and `isReadOnlyAttribute` function to hide elements on `assessment-info-pane` component
- Extended `noItemsText` to show message only in isSemiRestrictedOnStatus case

# Sanity checklist




# PR Review checklist
